gobblin-utility/src/main/java/org/apache/gobblin/util/filesystem/PathAlterationListener.java

```diff
@@ -43,4 +43,6 @@ public interface PathAlterationListener {
   void onDirectoryDelete(final Path directory);
 
   void onFileDelete(final Path path);
+
+  void onCheckDetectedChange();
```

Comment:
javadoc would clarify... it's not immediately intuitive what "check detected change" is, in the way that "directory delete" is familiar.